lib/index.js

```diff
@@ -262,6 +262,23 @@ class Study extends EventTarget {
         die();
       }).bind(this)
     )
+    this.on(
+      "uninstall-custom-reason", (function (reason) {
+        prefSvc.get('shield.debug') && console.log("uninstall-custom-reason", this.states);
+        this.states.push("uninstall-custom-reason");
+        if (this.flags.dying) {
+          this.final();
+          return;
+        }
+        this.flags.dying = true;
+        this.report(merge({}, telemetrySubset(this.config), {study_state: reason}), "shield");
+        let that = this;
+        generateTelemetryIdIfNeeded().then(()=>that.showSurvey(reason));
+        try {this.cleanup()} catch (err) {/*ok*/} finally { /*ok*/ }
```

Comment:
Maybe a follow-up issue since I see it's done in many other places in the code, but silently swallowing errors feels dirty.